Create wrap definition for mt32emu library

dreamer opened the pull request:
The simplest working version. Munt's project mt32emu library provides
several additional features (e.g. switching between several APIs or
replacing internal resampling library with libsoxr or libsamplerate),
but it's out of scope for the first version of meson wrap.

This buildsystem hardcodes MT32EMU_EXPORTS_TYPE to 3, which turns on
both C and C++ interface variants of the library (this is the default
behaviour of upstream buildsystem - see generated config.h file or
mt32emu README.md for the details).

eli-schwartz commented on mt32emu/src/meson.build:
```diff
@@ -0,0 +1,63 @@
+  mt32emu_c_header_map += { header : files('c_interface' / header) }
+endforeach
+
+mt32emu_sources = files([
```
`files()` does not need to accept one array, it can be `files('one.c', 'two.c')`

dreamer replied:
Thanks! Fixed :)

dreamer commented:
I know the review guidelines suggest amending and force-pushing to the review - I pushed a fixup commit instead (to make it easier to see what did I change). When review will be finished I'll squash all fixup commits together and do final force push.

I haven't seen other wraps writing documentation in their readme files… but perhaps it would be more useful to send it as a separate PR, for master branch (I hope there's no technical issues with having user instructions in that file?).

eli-schwartz commented:
The technical instructions in the readme seem to be wrong, though, provide is supported since 0.55, not 0.56, you don't need the `fallback:` kwarg unless `required: false` or < 0.55

The entire section could be replaced by a link to https://mesonbuild.com/Wrap-dependency-system-manual.html#provide-section

dreamer commented:
I fixed the version to say 0.55.0 - indeed, it works correctly with 0.55.3 on my system. I discovered it only because GitHub CI machines have Ubuntu 20.04, which uses Meson 0.53 - that's why I decided to document it.

Indeed, I could replace it with that link - but it leads to documentation for wrap provide section, so it's non-obvious for new users.

jpakkane commented on readme.txt:
```diff
@@ -1,3 +1,18 @@
+
+For meson < 0.55.0 wrap fallbacks need to be explicitly enabled:
+
+    mt32emu_dep = dependency('mt32emu', fallback : ['mt32emu', 'mt32emu_dep'])
```
There is no real harm in having this, but do note that it is only shown in Github repo view when someone explicitly switches to this branch. So not that many people. :)

dreamer replied:
Ah, right. Once the wrap will become available, I'll create a new PR, with readme.txt update only, targetted for master branch.

eli-schwartz reviewed (approved):
Fixups LGTM.

> When review will be finished I'll squash all fixup commits together and do final force push.

ok

dreamer committed:
Create wrap definition for mt32emu library

The simplest working version. Munt's project mt32emu library provides
several additional features (e.g. switching between several APIs or
replacing internal resampling library with libsoxr or libsamplerate),
but it's out of scope for the first version of meson wrap.

This buildsystem hardcodes MT32EMU_EXPORTS_TYPE to 3, which turns on
both C and C++ interface variants of the library (this is the default
behaviour of upstream buildsystem - see generated config.h file or
mt32emu README.md for the details).

dreamer commented:
> Fixups LGTM.
> 
> > When review will be finished I'll squash all fixup commits together and do final force push.
> 
> ok

Done :) Thank you for the review and help with resolving the problems earlier!

eli-schwartz commented:
```
$ mesonwrap review mt32emu --pull-request 1
Inspecting project mt32emu, branch 2.4.2.
Repo name valid: ✓
Branch name valid: ✓
Target branch is not master: ✓
Has readme.txt: ✓
Has LICENSE.build: ✓
Has upstream.wrap: ✓
Repo contains only buildsystem files: ✓
upstream.wrap has [wrap-file] section: ✓
upstream.wrap does not have [wrap-git] section: ✓
upstream.wrap has [provide] section: ✓
upstream.wrap has directory: ✓
upstream.wrap directory has no path separators: ✓
upstream.wrap has source_url: ✓
upstream.wrap has source_filename: ✓
upstream.wrap source_filename has no path separators: ✓
upstream.wrap has source_hash: ✓
upstream.wrap has source_url with version substring: ✗
```

meh, underscores in the version tag. :(

eli-schwartz commented:
`--allow-url-without-version` passes all checks and builds libmt32emu.so, double-checked via Mk.1 eyeball that the versions are aligned.
